fs/fs.go

```diff
@@ -409,7 +410,8 @@ func (fs *filesystem) Unmount(ctx context.Context, mountpoint string) error {
 		fs.layerMu.Unlock()
 		return fmt.Errorf("specified path %q isn't a mountpoint", mountpoint)
 	}
-	delete(fs.layer, mountpoint) // unregisters the corresponding layer
+	delete(fs.layer, mountpoint)                    // unregisters the corresponding layer
+	commonmetrics.CleanupLayerData(l.Info().Digest) // cleanup metrics data for the corresponding layer
```

Comment:
There is no guarantee that layer digests aren't duplicate among mountpoints. So this can unexpectedly cleanup a recorder being used by other mountpoints. 

Instead of managing recorders using map globally, could we record the read timestamp in each `layer.Layer` object and expose it via `layer.Layer.Info()`?

Reply:
Corrected.